docs/sources/writing-guide/shortcodes/index.md

```diff
@@ -143,6 +143,127 @@ This shortcode inserts a lists of links to the page's subpages and includes the
 {{</* section withDescriptions="true"*/>}}
 ```
 
+## `docs/reference` shortcode (option A)
+
+The `docs/reference` shortcode offers more flexible linking than the Hugo builtin `relref` shortcode. 
+It allows you to specify your link destination or destinations in one part of the file (usually at end of the file, like a footer) while using the link label in the body text, like so:
+
+**Body text:**
+```markdown
+I get more traffic from [Google][1] than from [Yahoo][2] or [MSN][3].
+```
+
+**Reference:**
+```markdown
+{{%/* docs/reference */%}}
+[1]: "http://google.com/" 
+[2]: "http://search.yahoo.com/" 
+[3]: "http://search.msn.com/"
+{{%/* /docs/reference */%}}
+```
+
+This shortcode is useful when:
+
+- You want to link to the same destination multiple times, because it saves you time formatting the link every time.
+- When content from one repository is published to more than one documentation set, because the `docs/reference` shortcode lets you specify appropriate links for each doc set.
+
+For example, alerting documentation is sourced from the Grafana repository and published to both Grafana documentation and Grafana Cloud documentation.
+Any link from alerting documentation to dashboard documentation, for example, which is also published to both Grafana and Grafana Cloud, needs to be specific as to which destination is appropriate for the product documentation.
+Grafana Cloud alerting documentation should link to Grafana Cloud dashboard documentation, and Grafana alerting documentation should link to Grafana dashboard documentation.
+
+The following content must be within the opening and closing tags of the `docs/reference` shortcode:
+
+ `label` - The label for the shortcode. This is the text that you enter in the link instead of an http link or file path.  
```

Comment:
Do we need to be careful to say that the label uses `[<LABEL>]` syntax whereas a "usual" link uses `(<DESTINATION>)` syntax.

Most importantly that the label is in square brackets and the destination uses parentheses.

nit: I'm not sure if we should be trying to be consistent with the command line placeholder style guidance: https://grafana.com/docs/writers-toolkit/write/style-guide/write-for-developers/#command-lines